include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -307,6 +307,9 @@
 #    define _ONEDPL_CPP20_RANGES_PRESENT 0
 #endif
 
+// Feature public macros
+#define ONEDPL_HAS_RANGES_API  _ONEDPL_CPP20_RANGES_PRESENT
```

Comment:
Which instruction are you going to add regarding the use of the macro? Are you going to recommend inclusion of `ranges` to have it or `onedpl_config` to make it available? 
I would recommend `ranges` because there is a dependency between `onedpl_config` and `common_config`, which should not be exposed. The downside is that `ranges` always be available for inclusion, but it currently has a static assert limiting libstdc++/libc++ versions.

Reply:
At least each test includes execution , which includes onedpl_config.h

Reply:
Probably, you didn't get the question. Let's return to it when [a more basic one](https://github.com/oneapi-src/oneDPL/pull/1457#discussion_r1735601257) is sorted out.

Reply:
Probably) you have a chance to express your question in simpler way :)

Reply:
I was wondering about how you would recommend users to get this macro, or which header they should include to obtain it.

Reply:
See https://github.com/oneapi-src/oneDPL/pull/1457#discussion_r1738391654